modules/services/xcompose.nix

```diff
@@ -0,0 +1,49 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  xComposeType = with types; either (attrsOf xComposeType) str;
```

Comment:
You can't have recursive types. Perhaps once `lazyAttrsOf` is available?

Reply:
Ok, I'll wait then

Reply:
`lazyAttrsOf` has been merged in NixOS/nixpkgs#70138.

Reply:
Thanks for the info